ntp-proto/src/filter.rs

```diff
@@ -118,7 +126,8 @@ impl LastMeasurements {
         system_precision: NtpDuration,
         frequency_tolerance: FrequencyTolerance,
     ) -> Option<(PeerStatistics, NtpInstant)> {
-        let dispersion_correction = (new_tuple.time - peer_time) * frequency_tolerance;
+        let dispersion_correction =
+            NtpInstant::abs_diff(new_tuple.time, peer_time) * frequency_tolerance;
```

Comment:
This is a mild change of behaviour compared to the old situation, although new_tuple.time should never be before peer_time. However, perhaps we should assert this? (at least in debug mode)